src/domains/User/User.repository.ts

```diff
@@ -0,0 +1,48 @@
+import { BadRequestException, Injectable, Logger } from '@nestjs/common';
+import { User } from './User.aggregate';
+import { EventStore } from '@peterdijk/nestjs-eventstoredb';
+import { Repository } from 'typeorm';
+import { User as UserProjection } from '../../infrastructure/read-models/user.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+
+@Injectable()
+export class UserRepository {
+  private readonly aggregate = 'user';
+
+  constructor(
+    private readonly eventStore: EventStore,
+    @InjectRepository(UserProjection)
+    private userProjectionRepository: Repository<UserProjection>,
+  ) {}
+  private logger = new Logger(UserRepository.name);
+
+  async findOneById(aggregateId: string): Promise<User> {
+    const user = new User(aggregateId);
+    const { events } = await this.eventStore.getEvents(
+      this.aggregate,
+      aggregateId,
+    );
+    user.loadFromHistory(events);
+    return user;
+  }
+
+  async findOneByUsername(username: string): Promise<User> {
+    try {
+      const { userId } = await this.userProjectionRepository.findOne({
```

Comment:
Is this the correct way in event-sourced pattern?
I now need the projection to look up data needed to proceed with a user-action flow.

How else could I look up an aggregate by some aggregate property?

In-memory stored repository of aggregates? Use redis for this?

Is this Akka?

Reply:
check cache in memory is done in other pr